annotation-file-utilities/src/annotator/find/Insertion.java

```diff
@@ -50,22 +50,66 @@
     protected Set<String> packageNames;
 
     /**
+     * The inner types to go on this insertion. See {@link ReceiverInsertion}
+     * for more details.
+     */
+    protected List<Insertion> innerTypeInsertions;
+
+    /**
      * Set of annotation names that should always be qualified, even
      *  when {@link getText(boolean, boolean)} is called with abbreviate true.
      */
     protected static Set<String> alwaysQualify = new LinkedHashSet<String>();
 
     /**
+     * Type to be inserted, if applicable; otherwise null.
+     */
+    protected Type type;
+
+    /**
+     * If true only the annotations from {@link type} will be inserted.
```

Comment:
Most often, only annotations get inserted.  (CloseParenthesesInsertion is the most obvious exception -- it inserts neither annotations nor types.)